schematic/store/synapse.py

```diff
@@ -788,18 +799,42 @@ def fill_in_entity_id_filename(
         # the columns Filename and entityId are assumed to be present in manifest schema
         # TODO: use idiomatic panda syntax
         if dataset_files:
+            all_files = self._get_file_entityIds(
+                dataset_files=dataset_files, only_new_files=False, manifest=manifest
+            )
             new_files = self._get_file_entityIds(
                 dataset_files=dataset_files, only_new_files=True, manifest=manifest
             )
 
-            # update manifest so that it contains new dataset files
+            all_files = pd.DataFrame(all_files)
             new_files = pd.DataFrame(new_files)
+
+            # update manifest so that it contains new dataset files
             manifest = (
                 pd.concat([manifest, new_files], sort=False)
                 .reset_index()
                 .drop("index", axis=1)
             )
 
+            manifest_reindex = manifest.set_index("entityId")
```

Comment:
there's no comment from line 819 - line 837. Could you add some comments to breakdown some of the steps?